ca/ca.go

```diff
@@ -573,13 +581,31 @@ func (ca *certificateAuthorityImpl) issuePrecertificateInner(ctx context.Context
 
 	serialHex := core.SerialToString(serialBigInt)
 
-	names := csrlib.NamesFromCSR(csr)
+	idents := identifier.FromCSR(csr)
+	var dnsNames []string
+	var ipAddresses []net.IP
+	for _, ident := range idents {
+		switch ident.Type {
+		case identifier.TypeDNS:
+			dnsNames = append(dnsNames, ident.Value)
+		case identifier.TypeIP:
+			ip := net.ParseIP(ident.Value)
+			if ip == nil {
+				return nil, nil, fmt.Errorf("parsing IP address: %s", ident.Value)
+			}
+			ipAddresses = append(ipAddresses, ip)
+		default:
+			return nil, nil, fmt.Errorf("evaluating identifier type: %s for %s", ident.Type, ident.Value)
+		}
+	}
```

Comment:
This chunk of code stands nicely on its own and has a clear interface: given a slice of idents, create two slices: one of DNS names and one of IP addresses. That suggests it should be its own function with its own unittests.

Also, since this is pretty much the inverse of `identifier.FromCSR`, I think it belongs in the identifier package. For instance, `identifier.SplitDNSAndIP`, returning `[]string, []net.IP, error`.

It's definitely a bummer that this function needs to return an error (from parsing the IP address), since the ultimate input (a parsed CSR) already has a parsed `net.IP` in it. By doing `identifier.FromCSR` followed by `identifier.SplitDNSAndIP`, we wind up parsing an IP, serializing, and re-parsing.